### DISCUSSION TITLE
Do the addons update on pushstate/replacestate/popstate?

### DISCUSSION THREAD

**Discussion by pufferfish101007** [ScratchAddons/ScratchAddons#621] [Help]
I'm making an addon (#620) which makes Scratch load a lot faster. To do that, rather than setting window.location.href to the next url, it uses window.history.pushstate. This means that the page won't reload.
For the browser's history navigation to work with this, the script which is making this all happen needs to be present on the new page as well. When testing this out just by running the userscript, the browser's history navigation changes the url but not the page content, suggesting that the userscript was not being loaded. Being the silly person I am, I didn't have any other addons enabled to see if they were loading or not.
So, the question is: do addons load/refresh when the window.history.pushState function is called? And window.onpopstate after using pushstate?
That's it. Big build up, small question :D
If they don't (I'm assuming they don't), would it be possible to add this feature to the addon loader api thingy? This would also eliminate the need for editor addons to check if they're on the editor from the project page. 
Otherwise, will the scripts I've appended to the bottom of the page content [here](https://github.com/pufferfish101007/ScratchAddons/blob/6a685e42fc6ded194435bda2e398bc16c374d378/addons/faster-loading/userscript.js#L48)  work?
Thanks.

**Comment by WorldLanguages** [comment]
> To do that, rather than setting window.location.href to the next url, it uses window.history.pushstate. This means that the page won't reload.

I don't think it's that easy. You can't just make a website that wasn't expected to work like that, magically work like that.

> So, the question is: do addons load/refresh when the window.history.pushState function is called? And window.onpopstate after using pushstate?

No.

**Reply by pufferfish101007** [reply]
> > To do that, rather than setting window.location.href to the next url, it uses window.history.pushstate. This means that the page won't reload.
>
> I don't think it's that easy. You can't just make a website that wasn't expected to work like that, magically work like that.

It does work. Trust me on that. It just doesn't work with scratchAddons.
> > So, the question is: do addons load/refresh when the window.history.pushState function is called? And window.onpopstate after using pushstate?
> 
> No.

Ok, thanks. Will running the background scripts as in background.html make the addons load/refresh or is it not as simple as that?
